### ISSUE TITLE
Bug in Rice compression of non-square image

### ISSUE THREAD

**Issue by BCowan12** [nom-tam-fits/nom-tam-fits#129]
I may be doing something wrong, but this is looking like a bug. When I create a Rice compressed file that isn't perfectly square, it fails to uncompress using funpack. The error is: FITSIO status = 414: error uncompressing image
number of table rows != the number of tiles in compressed image

My test application writes both a compressed and normal file. If I use fpack to compress the normal file, and compare the headers to the nom.tam compressed file, I notice that NAXIS2 is not the same as ZNAXIS2 in the nom.tam file, but they do match in the fpack compressed file. If something is causing the axes to switch in the nom.tam file, I guess that would explain why it works OK with the square image.

Here's my code. I'd really appreciate it if someone would let me know where I'm going astray.
`
```
import java.util.Arrays;

import nom.tam.fits.*;
import nom.tam.fits.compression.algorithm.rice.RiceCompressOption;
import nom.tam.fits.header.Compression;
import nom.tam.image.compression.hdu.*;
import nom.tam.util.BufferedFile;

public class FitsTest
{
    public static void main(String[] args)
    {
        try
        {
            short[][] imageArr = makeImage();

            // Create a RICE compressed FITS
            BasicHDU<?> hdu = FitsFactory.hduFactory(imageArr);
            CompressedImageHDU cHdu = CompressedImageHDU.fromImageHDU((ImageHDU)hdu);

            cHdu.setCompressAlgorithm(Compression.ZCMPTYPE_RICE_1)
            .setQuantAlgorithm((String)null)
            .getCompressOption(RiceCompressOption.class)
            .setBlockSize(32);
            cHdu.compress(); 

            Fits fits = new Fits();
            fits.addHDU(cHdu);

            fits.setChecksum();
            BufferedFile bf = new BufferedFile("compressed.fits.fz", "rw");
            fits.write(bf);
            bf.close();

            // Use same image to make uncompressed FITS
            BasicHDU<?> hdu2 = FitsFactory.hduFactory(imageArr);

            Fits fits2 = new Fits();
            fits2.addHDU(hdu2);

            fits2.setChecksum();
            BufferedFile bf2 = new BufferedFile("uncompressed.fits", "rw");
            fits2.write(bf2);
            bf2.close();

        } catch (Exception ex)
        {
            ex.printStackTrace();
        }
    }

    // Create a simple image arrays with contrast bands
    private static short[][] makeImage()
    {
        short[][] arr = new short[300][550];

        //short[][] arr = new short[300][300]; // Works fine if square

        for (int x = 100; x < 200; x++)
        {
            Arrays.fill(arr[x], 5, arr[x].length - 5, Short.MAX_VALUE); 
        }

        return arr;
    }
}
```
`

Thanks,
  Bruce

**Comment by BCowan12** [comment]
In further investigation, nom.tam code is unable to uncompress the non-square Rice compressed file that it created in my earlier example program.  It throws the exception:

`
java.lang.ArrayIndexOutOfBoundsException: 550
        at nom.tam.image.compression.tile.TileDecompressorInitialisation.init(TileDecompressorInitialisation.java:73)
        at nom.tam.image.compression.tile.TileDecompressorInitialisation.init(TileDecompressorInitialisation.java:43)
        at nom.tam.image.tile.operation.AbstractTiledImageOperation.createTiles(AbstractTiledImageOperation.java:128)
        at nom.tam.image.compression.tile.TiledImageCompressionOperation.read(TiledImageCompressionOperation.java:214)
        at nom.tam.image.compression.hdu.CompressedImageData.getUncompressedData(CompressedImageData.java:88)
        at nom.tam.image.compression.hdu.CompressedImageHDU.getUncompressedData(CompressedImageHDU.java:180)
        at nom.tam.image.compression.hdu.CompressedImageHDU.asImageHDU(CompressedImageHDU.java:140)
`

The same code successfully uncompresses a non-square file that was compressed using fpack.

**Comment by ritchieGitHub** [comment]
Sorry for the delay ... I was on a very long trip -> offline for a few Months ;-)
I will try to restart the comming weekend

**Comment by attipaci** [comment]
Hi I'm taking over maintaining the repo from Ritchie. I'm really not familiar with the work Ritchie did on the compression support of the library, so I cannot offer much help in fixing this. However, feel free to poke around on a fork of this repo, and if you find a fix, and create a pull request, I'll be happy to integrate it into the mainline releases...

**Comment by attipaci** [comment]
I will close this ticket on account of inactivity. Feel free to reopen if the issue is still present

**Comment by attipaci** [comment]
Reopening, since this came up again in #268. However let the newer issue track any progress from here on...

**Comment by attipaci** [comment]
I think this annoying issue is fixed at last in the above PR (#318).
